### PR TITLE
Put the Start On home setting behind a feature flag.

### PR DESCRIPTION
**Pull request by Amejia481**
Related conversation https://github.com/mozilla-mobile/fenix/issues/19876#issuecomment-888445331

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/HomeActivity.kt [hunk 1]
```diff
@@ -220,7 +220,7 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
             shouldNavigateBrowserFragmentOnCouldStart(savedInstanceState)
         ) {
             navigateToBrowserOnColdStart()
-        } else {
+        } else if (FeatureFlags.showStartOnHomeSettings) {
```
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/HomeActivity.kt:223]: Can we use `shouldStartOnHome()` here instead of this?

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/HomeActivity.kt:223]: Good point, we can but I would like to make explicit that we have to remove this check when removing the feature flag. As internally `shouldStartOnHome()` does a read from disk, that was already performed in the if condition statement above.

Comment (eliserichards) [reply app/src/main/java/org/mozilla/fenix/HomeActivity.kt:223]: That makes sense 👍

**Review by eliserichards — APPROVED** [review]
Lookin' good! 🚢

**Comment by Amejia481** [comment]
@Mergifyio backport releases_v91.0.0

**Pull request merged**